debug/rotate.html

```diff
@@ -0,0 +1,129 @@
+            background-color: #fff;
+            padding: 5px 0;
+            text-indent: 10px;
+            white-space: nowrap;
+            text-shadow:
+               -1px -1px 0 #fff,
+                1px -1px 0 #fff,
+                -1px 1px 0 #fff,
+                 1px 1px 0 #fff;
+        }
+    </style>
+</head>
+
+<body>
+<div id='map'></div>
+<div id='checkboxes'>
+  <input id='pitch-checkbox' name='allow-pitch' type='checkbox' checked> <label for='pitch-checkbox'>pitch on rotate</label>
+</div>
+
+<div id='buffer' style="display:none">
+    <em>Waiting for data...</em>
+</div>
+
+<script src='/dist/mapbox-gl-dev.js'></script>
+<script src='/debug/access-token-generated.js'></script>
+
+<script>
+function initMap(pitchOnRotate) {
+    if (window.map && window.map.destroy) {
+        window.map.destroy();
+    }
+
+    var map = window.map = new mapboxgl.Map({
+        container: 'map',
+        zoom: 12.5,
+        center: [-77.01866, 38.888],
+        style: 'mapbox://styles/mapbox/streets-v9',
+        hash: true,
+        pitchOnRotate: pitchOnRotate
+    });
+
+    map.addControl(new mapboxgl.Navigation());
+    map.addControl(new mapboxgl.Geolocate());
+    map.addControl(new mapboxgl.Scale());
+
+    map.on('load', function() {
+        map.addSource('geojson', {
+            "type": "geojson",
+            "data": "/node_modules/mapbox-gl-test-suite/data/linestring.geojson"
+        });
+
+        map.addLayer({
+            "id": "route",
+            "type": "line",
+            "source": "geojson",
+            "paint": {
+                "line-color": "#EC8D8D",
+                "line-width": {
+                    "base": 1.5,
+                    "stops": [
+                        [
+                            5,
+                            0.75
+                        ],
+                        [
+                            18,
+                            32
+                        ]
+                    ]
+                }
+            }
+        }, 'country-label-lg');
+
+        var bufferTimes = {};
+        map.on('tile.stats', function(bufferTimes) {
+            var _stats = [];
+            for (var name in bufferTimes) {
+                var value = Math.round(bufferTimes[name]);
+                if (isNaN(value)) continue;
+                var width = value;
+                _stats.push({name: name, value: value, width: width});
+            }
+            _stats = _stats.sort(function(a, b) { return b.value - a.value }).slice(0, 10);
+
+            var html = '';
+            for (var i in _stats) {
+                html += '<div style="width:' + _stats[i].width * 2 + 'px">' + _stats[i].value + 'ms - ' + _stats[i].name + '</div>';
+            }
+
+            document.getElementById('buffer').innerHTML = html;
+        });
+    });
+}
+
+initMap(true);
+
+document.getElementById('pitch-checkbox').onclick = function() {
+    initMap(!!this.checked);
+};
+
+</script>
+
+</body>
+</html>
```

Comment:
I created a new page because `pitchOnRotate` can't be mutated at runtime and requires creating a new Map. The other values can be mutated in place.